pkg/tfsandbox/details.go

```diff
@@ -232,6 +232,30 @@ func (p *Plan) Outputs() resource.PropertyMap {
 	return outputs
 }
 
+func (p *Plan) HasChanges() bool {
```

Comment:
Need to think on this again. Before the change the moduleState resource would not change unless inputs changed. That's not entirely accurate with respect to TF model - for example, tofu can detect drift. That might be a change independent of input changes. In this case, to write the tofu refreshed state into Pulumi the module state resource needs to decide an update plan.